pool/connection_pool.go

```diff
@@ -170,7 +173,9 @@ func ConnectWithOpts(addrs []string, connOpts tarantool.Opts, opts Opts) (*Conne
 	connPool.state.set(connectedState)
 
 	for _, s := range connPool.addrs {
-		go connPool.controller(s)
+		controllerCtx, cancel := context.WithCancel(context.Background())
+		connPool.ctxCancels = append(connPool.ctxCancels, cancel)
```

Comment:
We could use a single context with a single cancel function here.

Reply:
Thanks, updated.